src/garage/_environment.py

```diff
@@ -1,6 +1,222 @@
+            episode.
+
+    """
+
+    def __init__(self,
+                 observation_space,
+                 action_space,
+                 max_episode_length=math.inf):
+        self._max_episode_length = max_episode_length
+        super().__init__(action_space, observation_space)
+
+    @property
+    def action_space(self):
+        """Get action space.
+
+        Returns:
+            akro.Space: Action space of the env.
+
+        """
+        return self.input_space
+
+    @property
+    def observation_space(self):
+        """Get observation space of the env.
+
+        Returns:
+            akro.Space: Observation space.
+
+        """
+        return self.output_space
+
+    @action_space.setter
+    def action_space(self, action_space):
+        """Set action space of the env.
+
+        Args:
+            action_space (akro.Space): Action space.
+
+        """
+        self._input_space = action_space
+
+    @observation_space.setter
+    def observation_space(self, observation_space):
+        """Set observation space of the env.
+
+        Args:
+            observation_space (akro.Space): Observation space.
+
+        """
+        self._output_space = observation_space
+
+    @property
+    def max_episode_length(self):
+        """Get max episode steps.
+
+        Returns:
+            int: The maximum number of steps that an episode
+
+        """
+        return self._max_episode_length
+
+
+class EnvStep(
+        collections.namedtuple('EnvStep', [
+            'env_spec', 'observation', 'action', 'reward', 'next_observation',
+            'env_info', 'step_type'
+        ])):
+    # pylint: disable=missing-return-doc, missing-return-type-doc, missing-param-doc, missing-type-doc  # noqa: E501
+    r"""A tuple representing a single step returned by the environment.
+
+    Attributes:
+        env_spec (garage.envs.EnvSpec): Specification for the environment from
+            which this data was sampled.
+        observation (numpy.ndarray): A numpy array of shape :math:`(O^*)`
+            containing the observation for the this time step in the
+            environment. These must conform to
+            :obj:`env_spec.observation_space`.
+            The observation before applying the action.
+            `None` if `step_type` is `StepType.FIRST`, i.e. at the start of a
+            sequence.
+        action (numpy.ndarray): A numpy array of shape :math:`(A^*)`
+            containing the action for the this time step. These must conform
+            to :obj:`env_spec.action_space`.
+            `None` if `step_type` is `StepType.FIRST`, i.e. at the start of a
+            sequence.
+        reward (float): A float representing the reward for taking the action
+            given the observation, at the this time step.
+            `None` if `step_type` is `StepType.FIRST`, i.e. at the start of a
+            sequence.
+        next_observation (numpy.ndarray): A numpy array of shape :math:`(O^*)`
```

Comment:
I don't think it makes sense for this type to have both `observation` and `next_observation` anymore (it only made sense when it was the same as `TimeStep`)

Please revise it to only have `next_observation`, which is the observation produced in response to `action`.